src/io.c

```diff
@@ -719,34 +719,41 @@ nc_write_clb_flush(struct nc_wclb_arg *warg)
 static ssize_t
 nc_write_clb(void *arg, const void *buf, uint32_t count, int xmlcontent)
 {
-    ssize_t ret = 0, c;
+    ssize_t ret = 0;
+    int c;
     uint32_t l;
     struct nc_wclb_arg *warg = arg;
 
     if (!buf) {
         c = nc_write_clb_flush(warg);
-        if (c == -1) {
+        /* check if c in a valid range */
+        assert((int) (ssize_t) c == c);
+        if (c < 0) {
             return -1;
         }
-        ret += c;
+        ret += (ssize_t) c;
 
         /* endtag */
         c = nc_write_endtag(warg->session);
-        if (c == -1) {
+        /* check if c in a valid range */
+        assert((int) (ssize_t) c == c);
+        if (c < 0) {
             return -1;
         }
-        ret += c;
+        ret += (ssize_t) c;
 
         return ret;
     }
 
     if (warg->len && (warg->len + count > NC_WRITE_CHUNK_SIZE_MAX)) {
         /* dump current buffer */
         c = nc_write_clb_flush(warg);
-        if (c == -1) {
+        /* check if c in a valid range */
+        assert((int) (ssize_t) c == c);
+        if (c < 0) {
             return -1;
         }
-        ret += c;
+        ret += (ssize_t) c;
     }
```

Comment:
In the `proxy` branch, which will soon be merged into `master`, return values of `nc_write_clb_flush()` and `nc_write_endtag()` were changed to `int` so these changes are no longer needed.